src/global.d.ts

```diff
@@ -150,6 +154,33 @@ interface RenderReadySetting extends SettingItemGeneric {
 	userscriptReference?: IUserscriptInstance
 }
 
+interface UserscriptRenderReadySetting extends SettingItemGeneric {
+	type: ValidTypes;
+
+	// for sel
+	opts?: string[];
+	cols?: number;
+
+	// for multisel
+	/** optDescriptions.length must equal opts.length! */
+	optDescriptions?: string[];
+
+	// for num
+	min?: number;
+	max?: number;
+	step?: number;
+
+	// the data
+	key: string;
+	changed: Function;
+
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any
+	value: any;
```

Comment:
this might be a valid use case for `any`, but i'll test your branch to see if the values of this cannot be somewhat sanely described using types. i really don't like using `any`

Reply:
We should at least use `unknown` and avoid `any` whenever possible. But like KraXen said, it's probably possible to make it more strict than that.